src/Symfony/Component/VarDumper/Cloner/Data.php

```diff
@@ -298,7 +298,7 @@ private function dumpItem(DumperInterface $dumper, Cursor $cursor, array &$refs,
         if (!$item instanceof Stub) {
             $cursor->attr = [];
             $type = \gettype($item);
-            if ($item && 'array' === $type) {
+            if ('array' === $type && $item) {
```

Comment:
this is probably not the only place where coercing `NAN` to `bool` can bite us, but it's the only placed caught by our test suite